### PR TITLE
doc updates for per-function guc setting support

### PR DESCRIPTION
**Pull request by lisakowen**
- updated the SET, CREATE and ALTER FUNCTION SQL command reference pages
- included new examples in the CREATE and ALTER FUNCTION pages

wasn't sure if i needed to update release notes or any feature pages?

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 1]
```diff
@@ -112,7 +113,16 @@ it. <codeph>SECURITY DEFINER</codeph> specifies that the function is
 to be executed with the privileges of the user that created it. The key
 word <codeph>EXTERNAL</codeph> is allowed for SQL conformance, but it
 is optional since, unlike in SQL, this feature applies to all functions
-not just external ones.</pd></plentry><plentry><pt><varname>definition</varname></pt><pd>A string constant defining the function; the meaning depends on the
+not just external ones.</pd></plentry>
+                <plentry>
+                    <pt><varname>configuration_parameter</varname></pt>
+                    <pt><varname>value</varname></pt>
+                    <pd>The <codeph>SET</codeph> clause causes the specified session run-time
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: Lots of passive voice here.  Maybe reword to "The SET clause applies a value to a session configuration parameter when the function is entered."

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml [hunk 2]
```diff
@@ -159,7 +169,16 @@ quote, and a dollar sign. Inside the dollar-quoted string, single quotes,
...
+$SomeTag$Dianne's horse$SomeTag$</codeblock>
+            <p>If a <codeph>SET</codeph> clause is attached to a function, the effects of a
+                    <codeph>SET LOCAL</codeph> command executed inside the function for the same
+                variable are restricted to the function; the configuration parameter's prior value
+                is still restored at function exit. However, an ordinary <codeph>SET</codeph>
+                command (without <codeph>LOCAL</codeph>) overrides the <codeph>CREATE
+                    FUNCTION</codeph>
+                <codeph>SET</codeph> clause, much as it would for a previous <codeph>SET
+                    LOCAL</codeph> command. The effects of such a command will persist after
+                function exit, unless the current transaction is rolled back.</p><sectiondiv id="section7"><b>Using Functions With Queries on Distributed Data</b><p>In some cases, Greenplum Database does not support using functions in
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_FUNCTION.xml]: "function exit" -> "the function exits"

File: gpdb-doc/dita/ref_guide/sql_commands/SET.xml [hunk 3]
```diff
@@ -27,6 +27,15 @@ SET [SESSION | LOCAL] TIME ZONE {<varname>timezone</varname> | LOCAL | DEFAULT}<
           <codeph>SET LOCAL</codeph> within a single transaction: the <codeph>SET LOCAL</codeph>
         value will be seen until the end of the transaction, but afterwards (if the transaction is
         committed) the <codeph>SET</codeph> value will take effect.</p>
+      <p>If <codeph>SET LOCAL</codeph> is used within a function that includes a
+          <codeph>SET</codeph> option for the same variable (see <codeph><xref
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/SET.xml]: "variable" -> "configuration parameter"

**Comment by lisakowen** [comment]
thanks for reviewing, @dyozie - i have incorporated your comments.
